### PR TITLE
[DDW-1143] Fix languages stories

### PR DESCRIPTION
**Pull request by VanessaPC**
This PR fixes the storybook library store to work with the Languages components to they display user selection items on the dropdowns. 
[Jira ticket
](https://input-output.atlassian.net/browse/DDW-1143)

## Todos
- [x] Build storybook with ` yarn storybook:build`
- [x] Run storybook `yarn storybook`
- [x] Find the Languages components and check both of them. 
- [x] If any of the listed stories after the fix is still broken then the new ticket is raised for fixing that story

Languages components were correctly working in the application, but storybook was not providing a store that could reflect how they were behaving in the app. Therefore, is not necessary to test this in Daedalus.

## Screenshots
 The following shows that different values have been chosen. 
Screenshot 2022-10-18 at 12 59 42
Screenshot 2022-10-18 at 12 59 47
Screenshot 2022-10-18 at 13 01 00
Screenshot 2022-10-18 at 13 01 05

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

- [ ] Make sure the dropdowns allow you to change the information 

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [ ] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [ ] All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [x] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [x] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with typescript types
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [x] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by miorsufianiohk** [comment]
Hi @VanessaPC ,

As discussed on [Slack](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1666108332499669?thread_ts=1664972665.111299&cid=GGKFXSKC6), we are seeing "Date format" not populated when language is switched between EN/JP. Here is the [video](https://drive.google.com/file/d/12r4vQ4sWDXZKnA5TcB12kmn2sKNrZig8/view?usp=sharing) for reference. Thank you.

**Review by gabriela-ponce — APPROVED** [review]

**Review by miorsufianiohk — APPROVED** [review]
Thanks for fixing this quickly @VanessaPC . LGTM. Great work! 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,7 +8,8 @@
 
 ### Chores
 
-- Fixed locale in Storybook ([PR 3062](https://github.com/input-output-hk/daedalus/pull/3062))
+- Fixed Language components in Storybook to have it's own mocked store ([PR 3063](https://github.com/input-output-hk/daedalus/pull/3063))
+- Fixed locale in Storybook ([PR 3042](https://github.com/input-output-hk/daedalus/pull/3042))
```
Comment (szymonmaslowski) [inline CHANGELOG.md:12]:
```suggestion
- Fixed locale in Storybook ([PR 3062](https://github.com/input-output-hk/daedalus/pull/3062))
```

File: storybook/stories/settings/language/Language.stories.tsx [hunk 2]
```diff
@@ -1,40 +1,77 @@
...
+    if (value === mockedLanguageState.currentLocale) {
+      store.set({
+        currentDateFormat: mockedLanguageState.currentDateFormat,
+      });
+    } else {
+      store.set({
+        currentDateFormat: 'YYYY年MM月DD日',
+      });
+    }
+  }
```
Comment (szymonmaslowski) [inline storybook/stories/settings/language/Language.stories.tsx:48]:
```suggestion
  if (id === 'locale') {
    const currentDateFormat = value === mockedLanguageState.currentLocale
      ? mockedLanguageState.currentDateFormat
      : 'YYYY年MM月DD日';
    store.set({
      currentDateFormat,
    });
  }
```

**Review by szymonmaslowski — CHANGES_REQUESTED** [review]
Just two small things. If the sugested code change will require retesting then we can skip it :)

File: CHANGELOG.md (hunk 1 above)
Comment (miorsufianiohk) [reply CHANGELOG.md:12]: @szymonmaslowski @VanessaPC We can retest this if required. No problem 😄

**Review by szymonmaslowski — APPROVED** [review]
Looks good!

**Review by przemyslaw-wlodek — CHANGES_REQUESTED** [review]

**Review by przemyslaw-wlodek — APPROVED** [review]
LGTM ❤️ Thanks for addressing our comments!

**Review by tomislavhoracek — APPROVED** [review]
💯

**Pull request merged**